param/parameterized.py

```diff
@@ -2049,43 +2276,54 @@ def watch(self_, fn, parameter_names, what='value', onlychanged=True, queued=Fal
 
         See help(Watcher) and help(Event) for the contents of those objects.
         """
+        if precedence < 0:
+            raise ValueError("User-defined watch callbacks must declare "
+                             "a positive precedence. Negative precedences "
+                             "are reserved for internal Watchers.")
+        return self_._internal_watch(fn, parameter_names, what, onlychanged, queued, precedence)
```

Comment:
This reads to me as "complain when you see negative precedences (as they should be internal only) which means `_internal_watch` gets called with positive precedences'! Not sure whether this is just a naming issue but the error contradicts what you the next line seems to be suggesting...

Reply:
Can't quite follow this objection, `_internal_watch` allows any precedence while the public API is restricted to positive.

Reply:
Sure, does seem like an implementation detail users don't need to know about but I don't actually think saying negative precedences are **reserved** for internal watchers implies that they are exclusively negative in any way.

Reply:
Is `_watch` already used? The underscore suggest that it isn't public API and therefore internal...